src/test/java/org/cryptomator/cryptofs/health/dirid/OrphanDirTest.java

```diff
@@ -0,0 +1,304 @@
+
+		try (var uuidClass = Mockito.mockStatic(UUID.class)) {
+			UUID uuid = Mockito.mock(UUID.class);
+			uuidClass.when(() -> UUID.randomUUID()).thenReturn(uuid);
+			Mockito.doReturn("aaaaaa").when(uuid).toString();
+
+			Assertions.assertThrows(FileAlreadyExistsException.class, () -> result.prepareCryptoFilesystem(pathToVault, cryptor, clearStepParentName));
+		}
+
+		Assertions.assertNotEquals(Constants.RECOVERY_DIR_ID, Files.readString(pathToVault.resolve("d/00/0000/1.c9r/dir.c9r"), StandardCharsets.UTF_8));
+		Assertions.assertTrue(Files.notExists(pathToVault.resolve("d/11/1111/2.c9r/dir.c9r")));
+		Assertions.assertTrue(Files.notExists(pathToVault.resolve("d/22/2222")));
+
+	}
+
+	@Test
+	@DisplayName("prepareCryptFileSystem() runs without error on existing stepparent")
+	public void testPrepareCryptoFSExistingStepParentDir() throws IOException {
+		Path existingRecoveryDirFile = cipherRoot.resolve("1.c9r/dir.c9r");
+		Files.createDirectories(existingRecoveryDirFile.getParent());
+		Files.writeString(existingRecoveryDirFile, Constants.RECOVERY_DIR_ID, StandardCharsets.UTF_8, StandardOpenOption.TRUNCATE_EXISTING, StandardOpenOption.WRITE, StandardOpenOption.CREATE);
+		Path cipherRecovery = dataDir.resolve("11/1111");
+		Files.createDirectories(cipherRecovery);
+
+		Path existingStepparentDirFile = cipherRecovery.resolve("2.c9r/dir.c9r");
+		Files.createDirectories(existingStepparentDirFile.getParent());
+		Files.writeString(existingStepparentDirFile, "aaaaaa", StandardCharsets.UTF_8, StandardOpenOption.TRUNCATE_EXISTING, StandardOpenOption.WRITE, StandardOpenOption.CREATE);
+		Path cipherStepparent = dataDir.resolve("22/2222");
+		Files.createDirectories(cipherStepparent);
+
+		String clearStepParentName = "step-parent";
+		Mockito.doReturn("000000").when(cryptor).hashDirectoryId(Constants.ROOT_DIR_ID);
+		Mockito.doReturn("111111").when(cryptor).hashDirectoryId(Constants.RECOVERY_DIR_ID);
+		Mockito.doReturn("222222").when(cryptor).hashDirectoryId("aaaaaa");
+
+		Mockito.doReturn("1").when(cryptor).encryptFilename(BaseEncoding.base64Url(), Constants.RECOVERY_DIR_NAME, Constants.ROOT_DIR_ID.getBytes(StandardCharsets.UTF_8));
+		Mockito.doReturn("2").when(cryptor).encryptFilename(BaseEncoding.base64Url(), clearStepParentName, Constants.RECOVERY_DIR_ID.getBytes(StandardCharsets.UTF_8));
+
+		try (var uuidClass = Mockito.mockStatic(UUID.class)) {
+			UUID uuid = Mockito.mock(UUID.class);
+			uuidClass.when(() -> UUID.randomUUID()).thenReturn(uuid);
+			Mockito.doReturn("aaaaaa").when(uuid).toString();
+
+			result.prepareCryptoFilesystem(pathToVault, cryptor, clearStepParentName);
+		}
+
+		Assertions.assertEquals(Constants.RECOVERY_DIR_ID, Files.readString(pathToVault.resolve("d/00/0000/1.c9r/dir.c9r"), StandardCharsets.UTF_8));
+		Assertions.assertEquals("aaaaaa", Files.readString(pathToVault.resolve("d/11/1111/2.c9r/dir.c9r"), StandardCharsets.UTF_8));
+		Assertions.assertTrue(Files.isDirectory(pathToVault.resolve("d/22/2222")));
+	}
```

Comment:
These tests share a lot of test-setup code. Maybe use a `@Nested` test with a common `@BeforeEach`?

Reply:
Fixed in 277c51e